main.js

```diff
@@ -1,2 +1,163 @@
+
+let test = 1;
+test++;
+test += 1;
+// 3
+console.log('7.5', test);
+
+test--;
+// 2
+console.log('7.7', test);
+
+test = !!test;
+// true
+console.log('7.10', test);
+
+let part8 = [2, 3, 5, 8];
+let value = 1;
+for (let i = 0; i < part8.length; i++) {
+    value *= part8[i];
+}
+console.log('8', value);
+
+let part9 = [2, 5, 8, 15, 0, 6, 20, 3];
+for (let i = 0; i < part9.length; i++) {
+    value = part9[i];
+    if (value > 5 && value < 10) {
+        console.log('9', value);
+    }
+}
+
+let part10 = [2, 5, 8, 15, 0, 6, 20, 3];
+for (let i = 0; i < part10.length; i++) {
+    value = part10[i];
+    if (value % 2 === 0 && value !== 0) {
+        console.log('10', value);
+    }
+}
+
+// lesson 3
+
+function palindrome(str) {
+    str = str.toLowerCase().replace(/\s/g, '').split(',').join('');
+    return str === str.split("").reverse().join("");
+}
+
+console.log('1', palindrome('Тринитраплазмаполиуритан'));
+console.log('1', palindrome('Утроба, к аборту'));
+
+
+function checkMin(a, b) {
+    if (a < b) {
+        return a;
+    } else {
+        return b;
+    }
+}
+
+function checkMinSecVer(a, b) {
+    return a < b ? a : b;
+}
+
+
+function checkMax(a, y) {
+    if (a > y) {
+        return a;
+    } else {
+        return y;
+    }
+}
+
+function checkMaxSecVer(a, y) {
+    return a > y ? a : y;
+}
+
+console.log('2.1', checkMin(2, -38));
+console.log('2.2', checkMax(56, 7));
+console.log('2.3', checkMinSecVer(2, 87));
+console.log('2.3', checkMaxSecVer(65, 8));
+
+
+function getRandom(min, max) {
+    return Math.floor(Math.random() * (max - min) + min);
+}
+
+const arr = [];
+for (let i = 0; i < 10; i++) {
+    arr[i] = getRandom(0, 101);
+}
+arr.length = 10
+for (let i = 0; i < arr.length; i++) {
+    arr[i] = arr[i].toString().replace('0', 'zero');
+}
+
+console.log('3', arr);
+
+
+// lesson 4
+
+const sum = (a) => {
+    return (b) => {
+        return a + b;
+    }
+}
+
+console.log(sum(5)(2));
+
+const colors = ['magenta', 'cyan', 'firebrick', 'springgreen', 'skyblue'];
+
+const t1 = document.getElementsByClassName('text1');
+const t2 = document.getElementsByClassName('Text2');
+const t3 = document.getElementsByClassName('Text3');
+let c = 0;
+let d = 0;
+let e = 0;
+
+let colorCh1 = function colorChange1() {
+
+    return function () {
+        t1.style.color = colors[c];
+        c++;
+        if (c === colors.length) {
+            c = 0;
+        }
+    }
+}
+
+function colorChange2() {
+
+    t2.style.color = colors[d];
+    d++;
+    if (d === colors.length) {
+        d = 0;
+    }
+}
+
+function colorChange3() {
+    t3.style.color = colors[e];
+    e++;
+    if (e === colors.length) {
+        e = 0;
+    }
+}
+
+for (let i = 0; i < colors.length; i++) {
+    colors[i].addEventListener('click', colorCh1());
+}
+
+for (let i = 0; i < colors.length; i++) {
+    colors[i].addEventListener('click', colorChange2());
+}
+
+for (let i = 0; i < colors.length; i++) {
+    colors[i].addEventListener('click', colorChange3());
+}
```

Comment:
Let's create one universal function. We can declare a counter either inside or behind our function. 
The function will return a function that must take a special "Event" object in its arguments. 
From this "Event" object we can get the current color of our target and equate it to the color in the array. 
Next, we increment our counter on the iteration. We need to make the coloring infinite. Then, if the value of the counter is greater than the length of the color array, we reset the counter to zero. 

Then, for each item, we need to attach an event handler with our function, but don't forget that they must work independently of each other.